client/my-sites/plans/current-plan/purchases-listing.jsx

```diff
@@ -124,7 +124,52 @@ class PurchasesListing extends Component {
 		return null;
 	}
 
-	getExpirationInfo( purchase ) {
+	getPurchaseByCurrentPlan() {
+		const { currentPlanSlug, purchases } = this.props;
+
+		if ( ! currentPlanSlug ) {
+			return null;
+		}
+
+		return find(
+			purchases,
+			purchase => purchase.active && purchase.productSlug === currentPlanSlug
+		);
+	}
+
+	getExpirationInfoForPlan( plan ) {
+		// No expiration date for free plan or partner site.
+		if ( this.isFreePlan( plan ) || isPartnerPurchase( plan ) ) {
```

Comment:
It seems wrong to run a check called on `isPartnerPurchase()` on something that isn't a purchase.. I am not sure, but I suspect this code never worked before.

Similarly, `this.isFreePlan()` is called here on a plan, but further down it's called on a purchase... looking at the code, that might actually work in practice though.

Bottom line: I think there are existing bugs here (that you didn't cause) and I'm not sure the best way to handle them; they just come into glaring relief when you split this up into separate functions for the plan and purchase expiration :)

Reply:
I was a little worried about this too. I've removed the `isPartnerPurchase()` check for plans but have still kept the the `isFreePlan()` check for purchases since as you said, that might still work in practice.